### PR TITLE
🐛 Don't overwrite existing babel plugins in unit test --coverage flag

### PR DESCRIPTION
**Pull request by jonathantyng-amp**
Update to https://github.com/ampproject/amphtml/pull/23388

The control flow for the `--coverage` flag overwrites existing babel plugins, specifically the `minify-replace` plugin that is used to define experiment constants. This caused `--coverage` tests to fail.

This PR changes the `coverage` flow so that it doesn't overwrite existing babel plugins.

### CODE REVIEW

**Comment by jonathantyng-amp** [comment]
cc @estherkim for review :)

This is a fix for something I missed in the last PR https://github.com/ampproject/amphtml/pull/23388

**Comment by estherkim** [comment]
Hmm, do we need the replace plugin for code cov reports? Can you link me to a log where they're failing?

**Comment by jonathantyng-amp** [comment]
> Hmm, do we need the replace plugin for code cov reports? Can you link me to a log where they're failing?

I believe so, otherwise the experiment constant will be undefined and will throw an error. Here's a log https://travis-ci.org/ampproject/amphtml/builds/562265581?utm_source=github_status&utm_medium=notification

File: build-system/tasks/runtime-test/runtime-test-base.js [hunk 1]
```diff
@@ -222,7 +222,8 @@ class RuntimeTestConfig {
         'report-config': {lcovonly: {file: `lcov-${testType}.info`}},
       };
 
-      const plugin = [
+      const newBabelifyConfig = Object.assign({}, this.babelifyConfig);
```
Comment (estherkim) [inline build-system/tasks/runtime-test/runtime-test-base.js]: Can we modify the plugin array before calling `Object.assign()`?

Comment (jonathantyng-amp) [reply build-system/tasks/runtime-test/runtime-test-base.js]: What do you mean exactly? Concating `plugin` array and `this.babelifyConfig['plugins']`?

Comment (estherkim) [reply build-system/tasks/runtime-test/runtime-test-base.js]: Yeah exactly, then using that new array in `Object.assign()`. Also I think you can use `this.babelifyConfig.plugins`

**Comment by jonathantyng-amp** [comment]
@estherkim ready for another review :)

**Pull request merged**